### PR TITLE
Bug 2009352: IR-253: add prometheus rules for image registry operations metrics

### PR DESCRIPTION
**Pull request by flavianmissi**
image

### CODE REVIEW

File: manifests/09-prometheus-rules-registry-operations.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+spec:
+  groups:
+  - name: imageregistry.operations.rules
+    rules:
+    - expr: |
+        label_replace(
+          label_replace(
+            sum by (operation) (imageregistry_request_duration_seconds_count{operation="BlobStore.ServeBlob"}), "operation", "get", "operation", "(.+)"
+          ), "resource_type", "blob", "resource_type", ""
+        )
```
Comment (flavianmissi) [inline manifests/09-prometheus-rules-registry-operations.yaml:15]: this feels kinda ugly but it's the only way I found of adding a new label (the outermost `label_replace`) and changing the value of a label (the innermost `label_replace`)

File: manifests/09-prometheus-rules-registry-operations.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      record: imageregistry:operations_sum:count
```
Comment (dmage) [inline manifests/09-prometheus-rules-registry-operations.yaml:16]: I think it should be `imageregistry:operations_count:sum`

Comment (flavianmissi) [reply manifests/09-prometheus-rules-registry-operations.yaml:16]: darn you're right, how did it end up like this

Comment (flavianmissi) [reply manifests/09-prometheus-rules-registry-operations.yaml]: fixed

File: manifests/09-prometheus-rules-registry-operations.yaml (hunk 1 above)
Comment (ricardomaraschini) [reply manifests/09-prometheus-rules-registry-operations.yaml:15]: Why do we need to rename the labels ? If it is necessary then why not use a simple label_replace:

```
label_replace(
     sum by (operation) (imageregistry_request_duration_seconds_count{operation="ManifestService.Put"}), "operation", "manifestCreate", "operation", "(.+)"
)
```

Comment (flavianmissi) [reply manifests/09-prometheus-rules-registry-operations.yaml:15]: I found that separating resource from operation (i.e manifest / get) improved the overall metric readability.

**Comment by flavianmissi** [comment]
I'm on it

File: docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+| Operation | Resource type  |
+| --------- | -------------- |
+| `get`     | blob\|manifest |
+| `create`  | blob\|manifest |
+
+## `imageregistry:imagestreamtags_count:sum`
+
+| Source     | Location         | Description                                                     |
+| ---------- | ---------------- | --------------------------------------------------------------- |
+| `imported` | openshift\|other | Image Stream Tags imported in 'openshift' or 'other' namespaces |
```
Comment (xiuwang) [inline docs/metrics.md:14]: The quotation marks for `other` may be misleading to think it named `other` namespace.

Comment (flavianmissi) [reply docs/metrics.md]: Good point - removed.

Comment (dmage) [reply docs/metrics.md]: I would keep quotation in the second column, but remove them from the description.

Comment (flavianmissi) [reply docs/metrics.md]: hmm not sure I follow - do you mean the second column should look like:
```
'openshift'|'other'
```
or maybe did you mean we should use backticks, like the first column:
```
`openshift`|`other`
```
?

Comment (dmage) [reply docs/metrics.md]: | Source     | Location         | Description                                                     |
| ---------- | ---------------- | --------------------------------------------------------------- |
| `imported` | `openshift`, `other` | Image Stream Tags imported in 'openshift' or other namespaces |

```
| Source     | Location         | Description                                                     |
| ---------- | ---------------- | --------------------------------------------------------------- |
| `imported` | `openshift`, `other` | Image Stream Tags imported in 'openshift' or other namespaces |
```

wdyt?

**Comment by flavianmissi** [comment]
I think https://github.com/openshift/cluster-image-registry-operator/pull/775 has to merge before we can create a backport for this one

**Comment by flavianmissi** [comment]
I mean https://github.com/openshift/cluster-image-registry-operator/pull/777 not 775

**Pull request merged**